Object is not subscriptable error fix

Itamai opened the pull request:
Would be nice if you can test it before merging just to be safe. 
test_main.py tests passed and also the string from the issue didnt throw any expections.
closes #359 
closes #248

The strings from the issues are these ones:
Now the first one stays in thinking loop and second one outputs "Failed to crack"
```
rAVBMAXAhABANBuBQATANAyBTAVAVBgAkAXBOA3ArA0A5BKAmAbApBmArA0A-AEB4AcApBRBkBBANBuBQAcAoBWAmB*ATB0BQAZAOARBmB1AtBgA6AZAvAHAhABAZBhBQAdANAGApABAbBtB0AWBJBBBGB5AJBgAwAWBHB<ADAMBtBrByAXBnBSBjAMBNBpBzADBNAWAuBRAlBuB6AcAKBSAsBBAYAqB5AZAtANBCAlAbBuB6AWBkAGA>AFAhBgAxAZAsARBsBBAjBjByADBNAWAuBRAlBuB6AcAKBSA7AHB-AuB6AYAtAhBjAFAfBfBQAZAOARBmB1AtBgA6AZAvAHAhABAFBxByAXB|AHAbAlAXBeByADBOABB>AUA-AxAaAHBGAHAZAwA-AyB0AZAtAdBDAFAXBgA6AWBIB2AuB9ApBpBQAcAOAmAGB5AjBgB5AIAOABB2B9ARBgAwADBMAxBLAxATBgB8ADBNAxBxABAjB0B4AZAkAGALA5ADBgA5ATBpBmALA1A|AnByAUBEAGAGBLB-AvB3ADBNARBLAtAFBgAwADBOACBGB*AFBgA5AXBMARBbAlAVBgB3AIAMAmA2B9ARBuAQASAuA<ADAhA<AuBQAcApBSBTAUBNBxByAUBEAGAuB9AjBgA3AWBIBCBEBBAHBkB7AZAkAGAbAVAXBvB6AXBIBWArAHBTBgAxAdAuA<ADAFATBuB3ADBOABB>AUA-AnByATBuACADAlAXBvB3ADBMARBeBVABBvB5AbAoBhBGBMA-AvB8AXBMASBbAVAjBvB0AWBHBtBcBBAfBgB2ATBtAWAuBlAXBiBQAZAuA3AbAVAXBvB0ATBtABBeBfB-AwB3AYAoBGALA5AJBgByALA|AEAGBLB-AyBwAXBfBHADA9AbBwB2AYAuAmAGBIBFBfBQAaAtAdBDAMBLBgBQAMAnATAxA*AhBgA7AVBuABB*ABAjBjByADBOAmArAxAFBcB5AUBkAGA2BYA-AHByAYBOAmALAIBFBvBQAcAoBGArAVAjBuBQAYAoBxBuBMB<AkB3AaAtAhB5ABAYAqB5AZAtANBCAlAbBuB6AWBkAHADAFAhBuBwAUBnBSBZAwA-AcB3AZAEAGAmB9AfBgBQAcAHBRBbAVAXBvB2AeAkAHA7AlAjBjBQAZAMASBbAtAjBqB4AIAOAHArAJATBkB5AVBNAWAuBcA-AuB3AUBJBCB7AFAfBgBQAbANAWAWBUA-A*A2AZAOASBZABAgAcBzAUBpARBLAtAFBtBQAaAtAhBbAxAlBfB0AWBHBtBDAUBFBtB5AVBtAxBuBHB-AqBzAIAJABB2BEBFBoBQASAuAHAbAQBVBuAFACAFBxB>AfB-AfB3ADBOAxBpABAlBuByADBNAXAhA3BUAEB6AIANAXAZABA<AgA2AbAoBhB5ABAFBuB6AYAsAmAeBlAhBjByAUBEAGAzAFABBvBQAdANAGALALB-AcBQAcAHBRBLARAFBtBQAdAoBWAeBwA-AdByADBMABBGBIBjBtBwATBpBBBrAQA-AdB8ADBOABB>AUA-AtByATBsABBLAJATBgBQAYAoBxBuBMBFBpB6AIANAxBxABA<AgA4AYAsAxBpABApBjByAWB|AGAeB9AZBmB0AWBHBtBDAFAjBgA0AYBOAyADAxA<A0B3AYBuA<AsBBAoAjByADBOAGA2BlAXBvBQAbAnBdBDAQBhBkB3AZAfBEAeB9AfBgB2ADBJAXADAIBlBoBQAaAuAyADAMBLBcB6AIANAXAhABALBcB5ADBMASADA1AZBtByAGBtAyBRA1ATBgB5AXBfBGAuB9AfBoBwAWBEAGAjAlAhBvB5AaAsAnArAMBNBqB3AIANAxBxABATBgB6AdAMASBTAHBTBgAwAXBfBGA2B*AbBqB5AUBsA<ADAMBZBgA6AXBoBWAuBcA-AnAhAWBoBiBjAVAXBvBQAaAMASBTAUATBgAxAWBoBiBjAVAXBvBQAaAMASBTAUAJBsAQAbAsAWAWBlAXBiBQAaAuA<ADAxAZBqB1ADBNABBGBtAFBgA5AZAsAWAjAFA|AnByADBIARBuBdATBkB5AVBFAdBCA1A<ApB8ADBMABBrAIBRBvB3AXBEAHADAQB|AnB0AXBoBGAGB5AJBgA4AYAsA2AWBFAJBgB5ADBMAWAuBQA-AyByATB|AHADAFAJBgBQAZAsABBGBMBZBtB5ADBNAhB2BXB-AwB5AUBkAEAeB9AfBgB2ADBJAXADAIBlBoBQAYAuAyADAMBLBgB0AXB|AGAjAVAHBcB6AWBOA<ADA1AZBfByAWBEAHAjAVArBvBWAIAMAWAuBQA-AcBQAcAnBRBLAEBBBjBQAZAIByBRAAAJBnB3AXBHBRBkBBANBrB5AYBtANB5ABApBkB2AbAEAHArA5ABBqB7AZAuAiADA1A<ApB8ADBOAxBrAIA-AuB0AYBMASBZABAhBvB0AWBNA<ADAlAXBgA6AaAMARBGBDB-AkB3AZAHBWAuBNAtBuAQAVAHBXATAlAZBwB5ADBOAhBrAEBhBkB3AWBJByADAhA<AxByADBMASBzA9ATBxByAUBEAGA2BUBFBtBQAdANAGApABAtBgBwAXBJByAcBBAhBqB2AUBuABBGB1AFBuBQAYAKBSADAFABBeB0AUBMARBuBLBTBgA5AWBoBRBrAMBNBoByAXBfBGA2B4A-ArB6AXBJBGA2BIBFBgAUAaAtAhBOBVABBvByAUBEAGA>AQBVBqB6AXB|AGALA5ADBgA6AaAMANBDAxANBmByAFBlAbBLBoAUASB0AZAuAmApABADBqByAXBfBGAGBLB-AeB3AWBsANBDAZAfBqB2AKBcBkAbA9AXBvB5AYAuAnAEBBAjBqBQAcANAyBDAQBTBcB5AIAMAmArAxANBgBzALAEAEAeB9AfBgB2ADBJAXADAIBlBoBQAaAuAyADA5AZBvBQAcAoBWAmB*ATB0BQAcAHBWAuBRAZBoBQAdAMASB>ALBXBgAkAYBEAGA>AFAhBgA5AbAoByBjAHB-AkB3AIAMASADA*ANBgBxAUBkAGA2BYA-AeB2AYAuA3AbAlABBcB2AIAJABBLAMBNBpBQAbANAXAjAVAfBcB6AdAuAmApABAHBtB3AWBkADAhAPB-A<AhAGBEAGAmBFARBkB3AZAfBGAGBLB-AqB7AZAuAiABADBbBwAYAIAPAWArAFAfBuBQAbAoBBBhA4A-ANB0ATBoBGALAEBDBgAmATBjB2AeBlAXBvB3ATBoByAcBBA<AgAmAYAuABBGB4A-ArB5AbAnBhBrAIBhBqB5AIAMAXAhABAQAcB2AXBMABBrA4AVBOB8AUBNAhBrAfB-A>A3AWBNABBrAdAFBgA0AWB|AFAzAlAfBiB0AWBIBWAJAwA-AnB3AWBoB2ArAQA-AwB4AIANAxBuBUA-AqBzAIAOABB>AUA-AoB3AXBHBNBDA9A|AuBxAYBuAmApABAYAcB6AaAtAdBDAYBZBtByAcAgB<ADANAZBpB5AUBsA3AjAVAjBwB5ALAEAGAzAEBZBoBQAYAkAEAeB9AfBgB2ADBJAXADAIBlBoBQAcAMAXAbAIB<AiByAGBEAGALA5ADBgAzAWBoBWAuBcA-AvB0AcAIByBrAdALBgA6AaAMANBDANANBvByAXBfBGA2BYA-AvB0AZAkAHA7A9AfBfBQAaAtAdBDANATBcB5AXBoBWAbAFATBgA2AaAuABBrAEB<AvB6AXBHBNBcBBADBkB5ATBoByBzAVAfBgByAIAOABB>AUA-AwB3AZANAyBrAJAjBcBxAbAMANBDAIBZBwB5AYAnBNBsBBAYAqB5AZAtANBCAlAbBuB6AWBkAGAbA9AVBgB5ADBMAiBTA9AVBgA5AUBsA3AjAlAZBpB5ADBGASAsB|AbBuAZAHB|AGALA5ADBgAYAGBBBTABA4AhBzAQAbAnBdBBAJADBgBQARAIBWAuBlA|AwB5ADBIAmA2B5AZBtB6AWBkAGArALB-AIBwAWBNAyBTAQBVBiAQAKAKABB>AUA-AAB8AdAOAmArA1AFBuBQAbAnBdBCAdAZBqByAIAMAWAuBQA-AAB7AaAtA<AEBBA|A0BQAQAoBWAbAVAfBqBWAIAOAyBTAlAjBvByAWB|AGAGB4A-A0AaADBIAkAZA4A-APB0AaAuAyADAJAZBqB1ADBNAXAZABA<AgA6AcAHBRBLAMBNBuByADBNAxBsBBAjBjByADBOABB>AVAZBtB8ADBNAxBxABAFBvB0AaAsA3AZAwA-AxByAXBKBSADA*AZBrB6AWBMAXARABADBwB5AaAtAhB5ABAjBjByADBKAmArA5A<AkB5AXBnBWAuBNAFBuAQAVANAGApABAHBkB5AcApB<ADAxANBpByADBNAxBxABAYAqB5AZAtANBCAlAbBuB6AWBlA<ABAJAYAqB5AZAtANBDAlAbBuB6AWBkAGAjA9ATBqB5AIAOA2AGBLB-AcB2AUBuA<AsB4A|AsAQAYAoBxBmBVAhBgAzAcAIBxBkBBA<AgA2AaAtAhBpABANBpBQAcAnBRBbAMBNBqB3AIAGASAsB|AbBuAZAHBABbBLBpAoAjByADBOA3AjAFAXBfBwAXBHB<ADANALBwB3AaAfBGA2BYA-AHB3AXBHBRBkBBASArB5AYBtANBDAQBhBgByAIAOA2AGB5ABBgBQAdANAGApAAAdB1AYAMAOAyADAlAhBgA5AZAuAHATA9ADBwBxAUBsA<ADAJAFBnB3AYBfBGAzA9AfBgA6AaANAyBbAUA-AkB3AdAMASBTAVAhBvByAUBFAdBCAIBFBeB6AaAtAxBuBHB-AxAXAMAmACAsBHBfBgAwAWBHB<ABA|AXBxAYALABBzAZABAHBtB3AWBkACATARAFBgAjAaAtAhBGBJAlBuBQAQAIBxBuB9AfBwB2ADBMASBhABAaAcB2AbApBnArA0A|AgAxAeAkAEAbAlABBgB5AbAfBGALAEBFBgAwAWBOA2A0BBAfBgB4AcAIBxBjAQBBBgByAIANAWAsBBAjBjByAVBuAiADAVArBcBxAYBEAGA2BEBNBiB0AWBHBWAcBBAHBqB5AbAlA<ADAFABBeB3AWBuAGALA5ANBgByAIAMAnAEBBAKApBzAWBNAXAbAgA-AxByAXBJB2AGB9AXBuBQAZAJBmA2B0A-AvB0AZAkADAJAfBdB0AQAdAOAmALA5AhBnBwAYBNAWA2B4A-AdB8ADBJACAsBBAkAcBxAVBoBGALA0AXBKAFAVAoBGArAEBFBgAxATBtAdBCAkA-AiByAYBEAHAbA9AVBgBfA<AFBBB>AVAfBgBQAYAuAmApABAVBcB3AeAkAHAzAFAfBkBwAYBNAWA2B5AhBgA3AUB|AHADAFAhBuBwAUBnBSBZABAZBhBQATANAyBTAVAVBgAkAXBOA3ArA0A-AcB7AYAtAWAeBFA|AnByAGBEAGATAQBjBgA6AaAMANBDA1A<AlB3AXBIBXAjAfB-AjBwAYBHBNBDAIBlBhBzAZAuAmArAQA-AcB2AdAMASBTAFAjBkB3AWB|AGAGB4A-AuB3AWBsANBDAZAZBtB2AGBEAGATAfB-AkB3AaAHBRBbAMBFBfBQAaAOARBmB9AlBtBWAIANAyBRABAfBcB3AZANAxBmBlAhBgByAIAOAxB2BEBDBuBQAdAoBGAGBNALBgAyAbAoBdB7ALB-AnB3AWBoByADAVAnBgB3AIAOA2AeBlAJBjB6AbAPASADAJAFBnB0AUBuAhBLAJATBgBXAIAJAWAxABAtBqB6ADBMAXATAUA-AiB3AVBtAhB5ABAjBqBQAdAuA2ApABA<AgA4AYAuA3AbAFAJBgBQAbAnBdBCAxAZBtByAWBkAEAGB*AhBwB2AGBEAHAGB9AlBgA3AZAsARBhABAjBqBQAYAHBNBDAIBlBtByADBOABB>AVAfBgBQAaAuA2AsBdAjBgAwAWBKBXAjAhANBpBzADBMARBmBJA<AtB5AYAuA3AbAlAXBiBQAaANAWAjARAFBpBQAaAtAdBDAMBLBgBQAbAtAWAjARATBgBQAbAnBdBDAMBFBzB6ALA|AEALAxATBgA6AaAMANBCAxAZBtByAWBkAEAGB*AhBwB2ADBMAxBrA5AFBtBwAYBNAyBTAHB-AqB3AIAOABB>AUA-AEB3AdAMASBTA5AFBvBQAdAMARBuBQA-AvB3ADBOAmArA*AFBcB6AIAOAHATAVADBgBzAaAtAhBrAQA-AeB0AdAtAhBWBHB-AcB5ADBNAhBrANAFBuB5ATBuAnAEBwA-AoBwAVBoBWAuBcA-AvB0AaAuAyADAMBLBgBQAZAIBXATAIBjBgA6AcAJBRBpABAJBgB3AZAuAmALAMBZBtBQAbAoBdBDAMBLBgBQASAtAiBjAVAfBpByAYBFAdBCAlAjBgA6AXBnBSBZABA<AgAyAaAsA3AjAlAZBpBwAXBKBSADA9AHBgA3AYBHBSBRAAAfBwAYAIAJABBLAMBNBpBQAdAoByBTARAhBsAQAYAoBxBmBJANBpByAUBEAHA7AlAjBjBQAYAkAGA>AFAXBfBzAdAtA<ADA9AHBgA2AWBnBBBrAwA-AuByAWBJBBBrA5ABBgBQAcApBCBTAQBBBvB6AXBHBSBZAwA-AvB3ADBMAxBrA5AFBtBwAYBMANBCAxAZBtByAWBkAEAGB*AhBwB2ADBOAxB>AlABBjBQAbANAxB2BtAhBgA5AZAsAXAbA9AXBcBxAbAMANBsBBAoAjByADBMAxBrA5AFBtBwAYBMARBhABAYAqB5AZAtANBCAlAbBuB6AWBkAGAGBHB-AvB0AZAuAmArAZAZBtByADBMAWAeBYB<A0B5ADBMAiBTAVAFBgAzAcAIBxBkBBAfBgB4AZAuABBGBMBNBqB3ALAEAGAGB5APBgBxAYBMARBhABALBwB2AWBpBSBRAwA-AqB5AIANAhB2B4AVBeB0AYAuAmALANAjBgB5AaAuA3AjAlABBgA7AWBpBmAjAHB-AgB6AYAgBbBAAAAAA4=


R hvv blf tzgsvi yvuliv nv...sfmtib...gviirurvw... Xofgxsrmt blfi yzyvh gl blfi yivzhg. Vnkvili Vnsbi srh nzixsvw srh ovtrlmh rmgl lfi ozmwh... Ozrw hrvtv gl vevib uligivhh uiln sviv gl gsv Yofv Nlfmgzrmh. Izyrw zmw izevmlfh, sv yrgvh zmw yrgvh zdzb. Nvm lu gsv Mligs, blf hgzmw zg gsv kivxrkrxv. Blfi prmth szev uzrovw blf, hl mld blf gfim gl gsv tlwh! Zmw bvg blf wl mlg kovzw? Blf wl mlg pmvvo gl wfhg blfi svzwh drgs zhs? Rmhgvzw blf dzro,  Dsb szev gsv tlwh ulihzpvm fh?  Dv nfhg ollp rmgl gsv girzoh dv uzrovw olmt ztl! Rm z grnv kzhhvw, lfi dliow rmgvigdrmvw drgs zmlgsvi gsilfts zm fksvzezo hxslozih xzoo gsv Xlmqfmxgrlm lu gsv Hksvivh... Gsv tlwh zooldvw fmslob ulixvh gl hork rmgl lfi wlnzrm. Gsv luuhkirmt lu gszg xzgzxobhn dzh gsv mvuvirlfh ulixv xzoovw nztrx... Bvg dv wrw mlg yzmrhs rg, rmhgvzw hgfwbrmt gsv erov zixzmv uli lfi kldvi zmw dvzogs! Zmw gsv nlmhgvih zg lfi wlli...gsv fmslob ivorxgh lu gsrh Xlmqfmxgrlm? ...gsv gilooh...gsv xlikhv vzgvih...gsv dvivdloevh? Wrw dv izrhv lfi hdliwh ztzrmhg gsvn? Li szev dv ozrw gsrh yfiwvm lm lgsvih? Lm hl-xzoovw drgxsvih? Hgizb xsrowivm gzftsg gsv dzbh lu ulfo hlixvib, gsvri ylwrvh nfgzgvw gsilfts yozhksvnlfh irgfzo. Hvmg gl urtsg nlmhgvih gslfts gsvb xlfow mlg wrhgrmtfrhs tllw uiln vero. Gsv uorxpvi lu sfnzmrgb olmt vcgrmtfrhsvw drgsrm gsvn. Bvh, gsvri mfnyvih szev wdrmwovw gsilfts gsv bvzih. Yfg z uvd hgroo ilzn lfi ozmwh, luuvirmt gsvri yollwb dlip uli xlrm. Gl gsrh wzb gsvb hsznv fh drgs gsvri evib vcrhgvmxv! Gsv Mligs yovvwh, uolttvw yb dzi. Gsv yzggovh ziv gsv tlwh' dsrk, xszhgrhvnvmg uli lfi hrmh! Zmw ovg fh mlg ulitvg gsv gviilih, gsv hxlfitvh uiln yvblmw lfi dliow! Gsv Drow Sfmg irwvh gsv hpb drgs vevib ufoo nllm! Gsv wzip izrwvih zywfxg lfi xsrowivm rmgl ozmwh fmpmldm! Hlnv hzb gsvb svizow z hvxlmw Xlmqfmxgrlm! Xzm dv xszig z xlfihv yzxp rmgl gsv ortsg? Droo dv urmw gsv hgivmtgs gl yzmrhs gsv nztvh uiln lfi prmtwlnh? Fmrgv zilfmw gsv dzings lu gsv Vgvimzo Uriv? Mrts rh gsv Grnv lu gsv Hdliw zmw gsv Zcv! Mlmv droo urtsg gsrh dzi rm lfi hgvzw! Mrts rh gsv Grnv lu Nzwmvhh zmw Wrhwzrm!
```

bee-san commented:
> Would be nice if you can test it before merging just to be safe.
> test_main.py tests passed and also the string from the issue didnt throw any expections.
> closes #359

Hey! Can you add the string in that GitHub issue as a test, or will that not decode?

bee-san commented:
@all-contributors please add @Itamai for code and bugs :)

Itamai commented:
> > Would be nice if you can test it before merging just to be safe.
> > test_main.py tests passed and also the string from the issue didnt throw any expections.
> > closes #359
> 
> Hey! Can you add the string in that GitHub issue as a test, or will that not decode?

i added it to the pull request but sadly first one stays in thinking loop.
As far as i know the thinking loop is another issue but i dont think that im experienced enough to work on that.
And second string outputs "Failed to crack"

bee-san commented:
> > > Would be nice if you can test it before merging just to be safe.
> > > test_main.py tests passed and also the string from the issue didnt throw any expections.
> > > closes #359
> > 
> > 
> > Hey! Can you add the string in that GitHub issue as a test, or will that not decode?
> 
> i added the strings to the pull request but sadly first one stays in thinking loop.
> As far as i know the thinking loop is another issue but i dont think that im experienced enough to fix that issue smiley
> And second string outputs "Failed to crack"

The 2nd string will work well, if you can just make it run (don't look for output) and Assert True, if we have the bug again in the future it'll crash the program :)

Itamai commented:
> > > > Would be nice if you can test it before merging just to be safe.
> > > > test_main.py tests passed and also the string from the issue didnt throw any expections.
> > > > closes #359
> > > 
> > > 
> > > Hey! Can you add the string in that GitHub issue as a test, or will that not decode?
> > 
> > 
> > i added the strings to the pull request but sadly first one stays in thinking loop.
> > As far as i know the thinking loop is another issue but i dont think that im experienced enough to fix that issue smiley
> > And second string outputs "Failed to crack"
> 
> The 2nd string will work well, if you can just make it run (don't look for output) and Assert True, if we have the bug again in the future it'll crash the program :)

added test for brandon :)

bee-san commented:
Thanks a lot for your contrib! <3
